test/jdk/java/awt/List/MouseDraggedOriginatedByScrollBarTest.java

```diff
@@ -24,41 +24,52 @@
 /*
  * @test
  * @bug 6240151
+ * @key headful
  * @summary XToolkit: Dragging the List scrollbar initiates DnD
- * @library /java/awt/regtesthelpers
- * @build PassFailJFrame
- * @run main/manual MouseDraggedOriginatedByScrollBarTest
+ * @run main MouseDraggedOriginatedByScrollBarTest
 */
 
+import java.awt.EventQueue;
 import java.awt.FlowLayout;
 import java.awt.Frame;
 import java.awt.List;
-import java.awt.event.MouseMotionAdapter;
+import java.awt.MouseInfo;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.event.InputEvent;
 import java.awt.event.MouseAdapter;
 import java.awt.event.MouseEvent;
+import java.awt.event.MouseMotionAdapter;
 
 public class MouseDraggedOriginatedByScrollBarTest {
-
-    private static final String INSTRUCTIONS = """
-            1) Click and drag the scrollbar of the list.
-            2) Keep dragging till the mouse pointer goes out the scrollbar.
-            3) The test failed if you see messages about events. The test passed if you don't.""";
+    static Frame frame;
+    static volatile Point loc;
+    static volatile List list;
+    static volatile int width;
 
     public static void main(String[] args) throws Exception {
-        PassFailJFrame.builder()
-                .title("MouseDraggedOriginatedByScrollBarTest Instructions")
-                .instructions(INSTRUCTIONS)
-                .rows((int) INSTRUCTIONS.lines().count() + 2)
-                .columns(35)
-                .testUI(MouseDraggedOriginatedByScrollBarTest::createTestUI)
-                .logArea()
-                .build()
-                .awaitAndCheck();
+        // Restrict test to XToolkit
+        boolean isXToolkit = Toolkit.getDefaultToolkit()
+                .getClass().getName().equals("sun.awt.X11.XToolkit");
+        if (!isXToolkit) {
+            System.out.println("The test is XAWT-only.");
+            return;
+        }
+
+        try {
+            createUI();
+            test();
+        } finally {
+            if (frame != null) {
+                EventQueue.invokeAndWait(() -> frame.dispose());
+            }
```

Comment:
Let's be consistent:
```suggestion
            EventQueue.invokeAndWait(() -> {
                if (frame != null) {
                    frame.dispose();
                }
            });
```